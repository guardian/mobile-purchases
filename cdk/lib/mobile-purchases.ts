import type { GuStackProps } from "@guardian/cdk/lib/constructs/core";
import { GuStack } from "@guardian/cdk/lib/constructs/core";
import type { App } from "aws-cdk-lib";
import path from "path";
import {CfnInclude} from "aws-cdk-lib/cloudformation-include";

export class MobilePurchases extends GuStack {
  constructor(scope: App, id: string, props: GuStackProps) {
    super(scope, id, props);

    // ---- Existing CFN template ---- //
    const yamlTemplateFilePath = path.join(__dirname, "../..", "cloudformation.yaml");
    new CfnInclude(this, "YamlTemplate", {
      templateFile: yamlTemplateFilePath,
    });
  }
}
