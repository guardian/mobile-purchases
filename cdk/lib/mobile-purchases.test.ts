import { App } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import { MobilePurchases } from "./mobile-purchases";

describe("The MobilePurchases stack", () => {
  it("matches the snapshot", () => {
    const app = new App();
    const stack = new MobilePurchases(app, "MobilePurchases", { stack: "mobile", stage: "TEST" });
    const template = Template.fromStack(stack);
    expect(template.toJSON()).toMatchSnapshot();
  });
});
