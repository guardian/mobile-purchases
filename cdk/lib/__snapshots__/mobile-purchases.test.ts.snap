// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The MobilePurchases stack matches the snapshot 1`] = `
{
  "Metadata": {
    "gu:cdk:constructs": [],
    "gu:cdk:version": "TEST",
  },
}
`;
